src/UI/pages/upload.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import NotUploader from '../components/shared/Notauthenticated';
+import Upform from '../components/upload/Upform'
+import { useAppSelector } from '../state/hooks';
+
+
+function upload()  {
+  const role = useAppSelector((state) => state.auth.roles);
+  
+  if (role.includes('uploader')){
```

Comment:
I think this can be made shorter with:
`return role.includes('uploader') ? <Upform /> : <NotUploader name="uploader" />`